ddtrace/api.py

```diff
@@ -39,18 +39,33 @@ def _downgrade(self):
         self._encoder = JSONEncoder()
         self._headers.update({'Content-Type': self._encoder.content_type})
 
+    def handle_agent_response(self, response):
+        """
+        Logs the different http status code of the http requests towards the agent
+        """
+        status_code = response.status
+
+        # client error code
+        if 400 <= status_code <= 499:
+            log.error("Client error: #{}".format(response.msg))
```

Comment:
we need some form of rate limiting here ... otherwise we could spam the log a lot.